designs/administrator/4.6/column-management/index.md

```diff
@@ -0,0 +1,53 @@
+---
+parent: Administrator
+version: 4.6
+---
+
+# Column Management
+
+## Behavior
+Column management will live in the toolbar as a column icon on the Pods page on desktop. It will be hidden on mobile. Clicking on the icon will open a modal that lists all column names with checkboxes.
+![column-management-toolbar](img/column-management-toolbar.png)
+If the user is looking at “all projects”, then the checkbox next to “name” as well as “namespace”  in the modal will be shown as checked and disabled. Hovering over it will bring up a tooltip that explains that the columns may not be edited.
+![column-management-disabled-check](img/column-management-disabled-check.png)
+The modal will have an informational alert by default that reads “You can select up to 7 columns”. If 7 columns have already been selected, then all checkboxes will be disabled. Once the user deselects a checkbox, the other checkboxes will return to default state (rather than disabled). 
```

Comment:
I suspect this was probably discussed, but was it decided that not letting the user get into an error state (too many columns selected) was preferable to just letting them check too many columns and have an error and not let them save until they deselect another one?

I feel like I might come to this dialog and just want to add the Labels column, but it's disabled. Other columns are also disabled but selected and the tooltip indicates they can't be edited (but for a different reason, because they are required.) I need to determine that the max number of columns are already selected and unselect one, so that I can then select the column I want.

Alternatively, if all the columns were enabled, I would check Labels and immediately get an error inline alert telling me I have too many columns selected, which wouldn't be that big of a leap for me to understand since I just checked one, so I uncheck a different one and now the error goes away and I can save.

Reply:
@itsptk I played around with an error state and we discussed that - the inline alert in the modal would possible out of sight depending on where the user is selecting columns and PF doesn't have an error state for a checkbox/table so we decided to just leave it in the info alert.

Also I spoke with our content strategist Abi about the type of alert and she suggested this information alert since technically it's not an "error", it's more of a constant informational alert that the user needs to be aware of.